src/flows/pipes/RawFluxEditor/view.tsx

```diff
@@ -39,18 +40,20 @@ const Query: FC<PipeProp> = ({Context}) => {
   const {queries, activeQuery} = data
   const query = queries[activeQuery]
 
-  const updateText = useCallback(
-    text => {
-      const _queries = queries.slice()
-      _queries[activeQuery] = {
-        ...queries[activeQuery],
-        text,
-      }
-
-      update({queries: _queries})
-    },
-    [update, queries, activeQuery]
-  )
+  const updateText = text => {
+    const _queries = queries.slice()
+    _queries[activeQuery] = {
+      ...queries[activeQuery],
+      text,
+    }
+
+    update({queries: _queries})
+  }
+  const updateTextDebounce = useCallback(debounce(updateText, 500), [
```

Comment:
same concern here. Last time we had issues with updating the flux editor between tabs (either input didn't work when a tab was open, or it started forgetting charaters), it was because of a sub component using stale state when updating, which it would do on rerenders, all red flags that needed to be routed out

Reply:
For reference, here's the issue without debounce. It looks like the text is persisted only once (i.e. the "updating" logs stop when I stop tying) but the re-rendering goes on forever

https://user-images.githubusercontent.com/6411855/109862412-404ede00-7c15-11eb-8ab5-2aaef5f185db.mov

Reply:
Same as above - this issue is fixed by switching libraries.